src/dmd/statementsem.d

```diff
@@ -201,6 +202,19 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
             s.exp = checkGC(sc, s.exp);
             if (s.exp.op == TOK.error)
                 return setError();
+
+            if (doCheckUnused())
+            {
+                if (AssignExp ae = s.exp.isAssignExp)
```

Comment:
This code can go inside the expression visitor in `visit(AssignExp)`

Reply:
visit(AssignExp exp) is long with many returns so I'm highly unsure where in I should inject my code.